### PR TITLE
Remove deprecated container injections and compiler passes

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ~~#21451 #21625 #21284 #22010~~ #22805 
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
rebase needed to see tests green

**Comment by chalasr** [comment]
done, thanks

File: src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php [hunk 1]
```diff
@@ -57,18 +56,6 @@ class Translator extends BaseTranslator implements WarmableInterface
      */
     public function __construct(ContainerInterface $container, MessageSelector $selector, $defaultLocale = null, array $loaderIds = array(), array $options = array())
     {
-        // BC 3.x, to be removed in 4.0 along with the $defaultLocale default value
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php]: you forgot to remove the default value

**Review by stof — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php (hunk 1 above)
Comment (chalasr) [inline src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php]: fixed

**Comment by nicolas-grekas** [comment]
missing CHANGELOG updates in components (please check any your other PRs also if any)

**Comment by chalasr** [comment]
CHANGELOG entries added

File: src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php [hunk 2]
```diff
@@ -55,20 +54,8 @@ class Translator extends BaseTranslator implements WarmableInterface
      *
      * @throws InvalidArgumentException
      */
-    public function __construct(ContainerInterface $container, MessageSelector $selector, $defaultLocale = null, array $loaderIds = array(), array $options = array())
+    public function __construct(ContainerInterface $container, MessageSelector $selector, $defaultLocale, array $loaderIds = array(), array $options = array())
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Translation/Translator.php]: it looks like the docblock is not up to date - and we could add a "string" type hint meanwhile here

File: src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md [hunk 3]
```diff
@@ -4,8 +4,13 @@ CHANGELOG
 4.0.0
 -----
 
- * Removed `ValidateWorkflowsPass`
  * The default `type` option of the `framework.workflows.*` configuration entries is `state_machine`
+ * [BC BREAK] removed `AddConsoleCommandPass`, `AddConstraintValidatorsPass`,
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/CHANGELOG.md]: i'm not sore we need all these bc break prefixes, I'd remove them

**Comment by chalasr** [comment]
@nicolas-grekas comments addressed

**Comment by nicolas-grekas** [comment]
@chalasr would you mind opening a PR to fix that doc block in 3.3 (I guess)?

**Comment by nicolas-grekas** [comment]
Thank you @chalasr.

**Comment by chalasr** [comment]
@nicolas-grekas see https://github.com/symfony/symfony/pull/22822

**Pull request merged**